src/server/yamlServer.ts

```diff
@@ -101,18 +102,23 @@ connection.onCompletion(
     }
 );
 
-function getSuggestions(rowIndex: number, fileUrl: URL) {
-    const autoCompleteTelemetryData = new TelemetryData(TelemetryConstants.AUTOCOMPLETE);
-    autoCompleteTelemetryData.addProperty(TelemetryConstants.SERVER, TelemetryConstants.YAML_SERVER);
+function getSuggestions(rowIndex: number) {
+    const telemetryData: IAutoCompleteTelemetryData = {
+        eventName: "autoComplete",
+        properties: {
+            server: 'yaml',
+        },
+        measurements: {},
+    };
     const portalAttributeKeyPattern = /(.*?):/; // regex to match text like adx_pagetemplateid:
     const matches = getEditedLineContent(rowIndex, editedTextDocument)?.match(portalAttributeKeyPattern);
     const completionItems: CompletionItem[] = [];
     if (matches) {
-        autoCompleteTelemetryData.addProperty(TelemetryConstants.KEY_FOR_COMPLETION, matches[1]);
+        telemetryData.properties.keyForCompletion = matches[1];
```

Comment:
should this be set after the next line, and use the value of the variable `keyForCompletion`?

Reply:
oh it's deliberately added that location. In telemetry we want to get the exact field on which auto-complete was used.
Otherwise adx_pagetemplateid and adx_pagetemplate will both log the same value.